### ISSUE TITLE
Can't install via composer due to version locks

### ISSUE THREAD

**Issue by PoetikDragon** [RWOverdijk/AssetManager#220]
New Laminas project based off the skeleton with the developer toolbar, doctrine, and doctrine migrations installed I get this:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install rwoverdijk/assetmanager 2.0.0
    - Installation request for symfony/console (locked at v5.1.0) -> satisfiable by symfony/console[v5.1.0].
    - Conclusion: remove laminas/laminas-mvc 3.1.1
    - Installation request for rwoverdijk/assetmanager ^2.0 -> satisfiable by rwoverdijk/assetmanager[2.0.0, 2.x-dev].
    - Conclusion: don't install laminas/laminas-mvc 3.1.1
    - rwoverdijk/assetmanager 2.x-dev requires zendframework/zend-mvc 2.* -> satisfiable by zendframework/zend-mvc[2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.1.0, 2.1.1, 2.1.2, 2.1.3, 2.1.4, 2.1.5, 2.1.6, 2.2.0, 2.2.0rc1, 2.2.0rc2, 2.2.0rc3, 2.2.1, 2.2.10, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 2.2.9, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.3.5, 2.3.6, 2.3.7, 2.3.8, 2.3.9, 2.4.0, 2.4.0rc1, 2.4.0rc2, 2.4.0rc3, 2.4.0rc4, 2.4.0rc5, 2.4.0rc6, 2.4.0rc7, 2.4.1, 2.4.10, 2.4.11, 2.4.12, 2.4.13, 2.4.2, 2.4.3, 2.4.4, 2.4.5, 2.4.6, 2.4.7, 2.4.8, 2.4.9, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.7.0, 2.7.1, 2.7.10, 2.7.11, 2.7.12, 2.7.13, 2.7.14, 2.7.15, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7, 2.7.8, 2.7.9].
    - don't install zendframework/zend-mvc 2.0.3|don't install laminas/laminas-mvc 3.1.1
```
(And it goes on for like 30+ lines about not installing the laminas-mvc package)

When I run composer why symfony/console:
```
doctrine/doctrine-module      4.0.2   requires  symfony/console (^5.0)
doctrine/doctrine-orm-module  3.0.3   requires  symfony/console (^5.0)
doctrine/migrations           2.2.1   requires  symfony/console (^3.4||^4.0||^5.0)
doctrine/orm                  v2.7.3  requires  symfony/console (^3.0|^4.0|^5.0)
```

And why laminas/laminas-mvc:
```
poetikdragon/pedigree            dev-master  requires  laminas/laminas-mvc (^3.0.1)
doctrine/doctrine-module         4.0.2       requires  laminas/laminas-mvc (^3.1)
laminas/laminas-developer-tools  1.3.2       requires  laminas/laminas-mvc (^2.7 || ^3.0.1)
laminas/laminas-mvc              3.1.1       replaces  zendframework/zend-mvc (self.version)
neilime/twbs-helper-module       v3.1.0      requires  laminas/laminas-mvc (^3.1)
```

I don't know why it can't resolve an installable package? Any ideas?

**Comment by PoetikDragon** [comment]
I managed to track it down to this issue: https://github.com/kriswallsmith/assetic/issues/886

Could you switch to `"assetic/framework": "dev-2.0/dev"` instead of `"kriswallsmith/assetic": "^1.4.0"` in composer.json?

**Comment by Ocramius** [comment]
You can try a local install with `"assetic/framework": "dev-2.0 as 1.99.99"` or such.

`composer why-not <package-name>` gives you more details on any conflicts.

**Comment by LukeTowers** [comment]
If you guys could test out assetic/framwork dev-2.0/dev and let me & @jaxwilko know how it works for you that would be great. We'd like to get it out as a stable release but we need people to test it with their projects.

**Comment by LukeTowers** [comment]
@larryb did `composer why-not <package>` ever work for you when you were having all those conflicts?

**Comment by PoetikDragon** [comment]
I am using assetic/framework dev-2.0/dev now. It is semi-successful for me, in that it stopped the conflict with composer and Laminas is able to find and load the AssetManager module.

However, none of my paths in the asset manager resolver configs are working, so I need to dig deeper into the problem.

**Comment by JaxWilko** [comment]
@PoetikDragon if at all possible any test cases or examples of errors would be really helpful :)

**Comment by PoetikDragon** [comment]
Found the problem! It was failing on checking if an asset was an instance of AssetInerface.

Changing `Assetic\Asset\AssetInterface` to `Assetic\Contracts\Asset\AssetInterface` everywhere in AssetManager fixed everything.

**Comment by LukeTowers** [comment]
@JaxWilko We should probably class_alias all of the classes that we moved so that it can be a drop in replacement, could you give that a try?

**Comment by JaxWilko** [comment]
@LukeTowers I'll take a look tomorrow morning

**Comment by PoetikDragon** [comment]
FYI I found a working fork of AssetManager: https://github.com/jield-webdev/laminas-assetmanager
It relies on this fork of Assetic: https://github.com/jield-webdev/assetic

**Comment by LukeTowers** [comment]
@PoetikDragon that fork of Assetic is just our fork (@assetic-php) but with the 2.0 branch merged into master.

**Comment by JaxWilko** [comment]
@PoetikDragon @LukeTowers I've just pushed some class aliasing for all the old interfaces to link to the new ones on the `2.0/dev` branch.

@PoetikDragon If you could try pulling in those changes via composer and letting us know if that fixes your problem that would be great

**Comment by BrunoSpy** [comment]
Any news on this issue ?

**Comment by LukeTowers** [comment]
We've just released the 2.0 release that addresses a number of existing issues and dependency updates, see https://github.com/assetic-php/assetic/releases/tag/v2.0.0 for more information! Please try it out and let us know if you have any issues!

**Comment by BrunoSpy** [comment]
Will https://github.com/jield-webdev/laminas-assetmanager be updated too ?

**Comment by LukeTowers** [comment]
@BrunoSpy see https://github.com/jield-webdev/laminas-assetmanager/pull/1

**Comment by BigMichi1** [comment]
i published a version using assetic/framework 2.0 to packagist, if you are interested in it check https://github.com/BigMichi1/assetmanager or https://packagist.org/packages/bigmichi1/assetmanager

**Comment by jlaudenbach** [comment]
Issue should be fixed with newly released version `3.0.0`.
